ResearchKit/Common/ORKTextChoiceCellGroup.m

```diff
@@ -107,6 +107,15 @@ - (void)didSelectCellAtIndex:(NSUInteger)index {
         }
     } else {
         touchedCell.selectedItem = !touchedCell.selectedItem;
+        if (touchedCell.selectedItem) {
+            ORKTextChoice *touchedChoice = [_helper textChoiceAtIndex:index];
+            for (NSNumber *num in [_cells allKeys]) {
+                ORKChoiceViewCell *cell = _cells[num];
+                if (cell != touchedCell && (touchedChoice.exclusive || (cell.selectedItem && [_helper textChoiceAtIndex:[num unsignedIntegerValue]].exclusive))) {
```

Comment:
Can we break picking the currently selected choice out into a separate line? Would make the logic easier to read.